File: Core/Entities/Favourites.cs
namespace Core.Entities;

public class Favourites
{
    public Favourites()
    {
    }

    public Favourites(string id)
    {
        Id = id;
    }

    public string Id { get; set; }
    public List<FavouriteFilm> FavouriteFilms { get; set; } = new List<FavouriteFilm>();
    public string ClientSecret { get; set; }
}

File: ClientApp/src/app/shared/models/favourites.ts
import * as cuid from 'cuid';

export interface FavouriteFilm  {
  id: number;
  movieTitle: string;
  pictureUrl: string;
  movieCountry: string;
  movieGenre: string;
}

export interface Favourites {
  id: string;
  favouriteFilms: FavouriteFilm[];
  clientSecret?: string;
}

export class Favourites implements Favourites {
  id = cuid();
  favouriteFilms: FavouriteFilm[] = [];
}
